test_common/harness/deviceInfo.cpp

```diff
@@ -29,44 +34,40 @@ static void *alloc_and_get_device_info( cl_device_id device, cl_device_info para
         throw std::runtime_error("clGetDeviceInfo failed\n");
     }
 
-    if (0 == size)
-        return NULL;
+    if (0 == size) return "";
```

Comment:
We should probably throw an exception here instead of testing against an empty string in all users.

Reply:
I'm not sure about this. While this might be fine for all current use cases (every path to this function will always result in a non-empty string on a correct OpenCL implementation), this might not always be true. For example, `get_device_il_version_string` could be used to detect the presence of an optional IL (i.e. empty string would be a valid, non-fatal result), or it might be valid for some future implementation to not support any extensions (thus validly returning an empty string). Also this helper function could be used for any of the other device queries, some of which also could produce empty strings.

Reply:
An empty string will still have the `NUL` character and be of size 1 ;).

Reply:
Good point 😳 

Done, PTAL.